src/backend/egl/surface.rs

```diff
@@ -4,22 +4,21 @@ use super::{ffi, native, EGLError, SurfaceCreationError, SwapBuffersError};
 use crate::backend::egl::display::EGLDisplayHandle;
 use crate::backend::graphics::PixelFormat;
 use nix::libc::c_int;
-use std::sync::Arc;
-use std::{
-    cell::Cell,
-    ops::{Deref, DerefMut},
-};
+use std::sync::{Arc, atomic::{AtomicPtr, Ordering}};
+use std::ops::{Deref, DerefMut};
 
 /// EGL surface of a given EGL context for rendering
 pub struct EGLSurface<N: native::NativeSurface> {
     pub(crate) display: Arc<EGLDisplayHandle>,
     native: N,
-    pub(crate) surface: Cell<ffi::egl::types::EGLSurface>,
+    pub(crate) surface: AtomicPtr<nix::libc::c_void>,
```

Comment:
We access the surface quite often, this should be much faster then a Mutex.
A little more difficult to handle, but manageable.